### PR TITLE
Add symbol to prevent breaking kubernetes plugin with jcasc

### PR DESCRIPTION
**Pull request by timja**
Fixes https://github.com/jenkinsci/configuration-as-code-plugin/issues/1337

cc @xuzhang3

### CODE REVIEW

File: src/main/java/com/microsoft/jenkins/containeragents/KubernetesCloud.java [hunk 1]
```diff
@@ -401,6 +402,7 @@ public static String getServiceNameWithoutOrchestra(String serviceName) {
     }
 
     @Extension
+    @Symbol("azureKubernetes")
```
Comment (gavinfish) [inline src/main/java/com/microsoft/jenkins/containeragents/KubernetesCloud.java]: Take care of the name here, there is a related one, [Azure Container Service (AKS) Plugin](https://github.com/jenkinsci/azure-acs-plugin).

Comment (timja) [reply src/main/java/com/microsoft/jenkins/containeragents/KubernetesCloud.java]: I can’t see a cloud in there, it just seemed to have a step called acsDeploy so should be fine unless I’m missing something

File: src/main/java/com/microsoft/jenkins/containeragents/KubernetesCloud.java [hunk 2]
```diff
@@ -401,6 +402,7 @@ public static String getServiceNameWithoutOrchestra(String serviceName) {
...
     public static class DescriptorImpl extends Descriptor<Cloud> {
         @Override
         public String getDisplayName() {
```
Comment (oleg-nenashev) [inline src/main/java/com/microsoft/jenkins/containeragents/KubernetesCloud.java]: Time to remove ACS from docs and UI? Should be a separate PR tho

File: src/main/java/com/microsoft/jenkins/containeragents/KubernetesCloud.java (hunk 1 above)
Comment (oleg-nenashev) [reply src/main/java/com/microsoft/jenkins/containeragents/KubernetesCloud.java]: I do cannot propose a better name which would deconflict plugins. Both focus the same feature, more or less. Suggestion below is a technical one which does not help too much

```suggestion
    @Symbol("azureKubernetesCloud")
```

Comment (timja) [reply src/main/java/com/microsoft/jenkins/containeragents/KubernetesCloud.java]: I would prefer not to include cloud in the same as it's part of the extension point.
i.e. for KubernetesCredentials the symbol would be kubernetes

**Comment by xuzhang3** [comment]
LGTM

**Comment by timja** [comment]
Would you be able to ship a release please @xuzhang3

**Comment by xuzhang3** [comment]
> Would you be able to ship a release please @xuzhang3

New version [v1.2.0](https://repo.jenkins-ci.org/releases/org/jenkins-ci/plugins/azure-container-agents/1.2.0/) has been released.
